base_location/tests/test_base_location.py

```
@@ -204,6 +204,14 @@ def test_company_onchange_state(self):
         self.company._onchange_state_id()
         self.assertEqual(self.company.country_id, self.company.state_id.country_id)
 
+    def test_partner_address_field_sync(self):
+        """Test that zip_id is correctly synced with parent of contact addresses"""
+        partner = self.env.ref("base.res_partner_address_1")
+        parent = partner.parent_id
+        self.assertEqual(partner.type, "contact")
```

You shouldn't test features outside the module, and this is also needed because of using demo data.

---

I don't think he's testing anything here. Such statements can be used simply to ensure that the state of an obj is what you expect and let the test fail if the original record changed. See above for the pro and con.

---

Any assert must be for testing module features. Requiring this for ensuring previous data state is a consequence of using demo data or not preparing correctly your test data. Please go for the other approach.

---

Look a direct problem I have found for asserting things that are not of the module itself:

https://github.com/OCA/l10n-spain/commit/a205e3f80148d871c9544b1106553d5e304c9d20#diff-3b8f173a5205124604ea5cef331b3e906af2e0a743921c59dd5d78a63ed3ca5cL95

This was re-testing things of the main module (because it's the expected thing), and when this main module changes due to OCA/bank-payment#860, then it breaks without being a problem of this module.